eclair-core/src/main/scala/fr/acinq/eclair/message/OnionMessages.scala

```diff
@@ -18,17 +18,23 @@ package fr.acinq.eclair.message
 
 import fr.acinq.bitcoin.Crypto.{PrivateKey, PublicKey}
 import fr.acinq.eclair.crypto.Sphinx
+import fr.acinq.eclair.io.MessageRelay.RelayPolicy
 import fr.acinq.eclair.wire.protocol.MessageOnion.{BlindedFinalPayload, BlindedRelayPayload, FinalPayload, RelayPayload}
 import fr.acinq.eclair.wire.protocol.OnionMessagePayloadTlv.EncryptedData
 import fr.acinq.eclair.wire.protocol.RouteBlindingEncryptedDataTlv._
 import fr.acinq.eclair.wire.protocol._
 import scodec.bits.ByteVector
 import scodec.{Attempt, DecodeResult}
 
+import scala.concurrent.duration.FiniteDuration
 import scala.util.{Failure, Success}
 
 object OnionMessages {
 
+  case class OnionMessageConfig(relayPolicy: RelayPolicy, timeout: FiniteDuration)
+
+  trait OnionMessageResponse
```

Comment:
A non-sealed trait doesn't let the compiler check for exhaustiveness in pattern matching.
Whenever you'll pattern-match on a trait, you should try to make it `sealed`.

I think this shows an issue with the architecture of this feature.
This trait should be the response trait for `Switchboard.SendMessage`: there should be one `case class` extending this trait for each possible thing that can happen when sending an onion message:

- we sent the message, and it's not expecting a response (yay!)
- we received a response (yay!)
- we couldn't send the message
- we sent the message, expected a response, but didn't get one

This trait (and its child case classes) should be defined in the actor that handles `SendMessage`.
I don't think it's the job of the `Switchboard` to do this mapping, I believe the `Postman` would be a better candidate for this.
The responsibilities of the `Postman` would then be:

- receiving `SendMessage` requests
- storing information about who to reply to if a response is expected
- asking the switchboard to actually send the onion message
- receiving a relay error, a response, or timing out -> crafting the corresponding `OnionMessageResponse` and sending it to its intended recipient actor

This feels like a cleaner separation of responsibilities (not overloading the switchboard with new responsibilities and somewhat weird anonymous short-lived actors).

This also removes the `SubscribeOnce` / `Unsubscribe` paradigm in favor of a simpler approach: the `SendMessage` request is that "subscription", and when that message has been completely handled, we remove it from the internal state of the `Postman`: no need for an external actor to call `Unsubscribe` at the right time (which could go wrong).

WDYT?

Reply:
Done